src/picomsg/src/routing_table/RoutingTable.cpp

```diff
@@ -0,0 +1,104 @@
+
+#include "RoutingTable.hpp"
+#include "logger.h"
+
+namespace picomsg {
+
+namespace detail {
+
+/*!*!*!*!*!*!*!
+ * RouteSPKey *
+ *!*!*!*!*!*!*/
+RouteSPKey::RouteSPKey(const struct routing_table_route_entry_sp_info& route_entry_sp_info)
+	: m_route_entry_sp_info(route_entry_sp_info)
+{ }
+
+/*!*!*!*!*!*!*!
+ * RouteEntry *
+ *!*!*!*!*!*!*/
+RouteEntry::RouteEntry(const struct routing_table_route_entry& route_entry)
+	: m_route_entry(route_entry)
+{ }
+
+const struct routing_table_route_entry& RouteEntry::to_raw() const {
+	return m_route_entry;
+}
+struct routing_table_route_entry& RouteEntry::to_raw() {
+	return m_route_entry;
+}
+
+} /* namespace detail { */
+
+RoutingTable::RoutingTable() {
+}
+
+RoutingTable::~RoutingTable() {
+}
+
+bool RoutingTable::init() {
+	return true;
+}
+
+std::unique_ptr<RoutingTable> RoutingTable::create() {
+	// Thank you C++
+	std::unique_ptr<RoutingTable> table { new(std::nothrow) RoutingTable{} };
+	if (! table) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to allocate %zu bytes for RoutingTable",
+					sizeof(RoutingTable));
+		return nullptr;
+	}
+
+	if (! table->init()) {
+		// Reason already logged @ function
+		return nullptr;
+	}
+
+	return table;
+}
+
+bool RoutingTable::add_route(const struct routing_table_route_entry& _route_entry) {
+	std::lock_guard _guard(m_lock);
+
+	if (! m_routes_by_sp.emplace(_route_entry.sp_info, _route_entry).second) {
+		return false;
+	}
+
+	return true;
+}
+
+bool RoutingTable::del_route_by_sp(const struct routing_table_route_entry_sp_info& route_entry_sp_info,
+								   bool& is_route_installed) {
+	std::lock_guard _guard(m_lock);
+
+	const detail::RouteSPKey route_sp_key(route_entry_sp_info);
+
+	const auto iter = m_routes_by_sp.find(route_sp_key);
+	if (m_routes_by_sp.end() == iter) {
+		return false;
+	}
+
+	is_route_installed = iter->second.to_raw().is_installed;
+	m_routes_by_sp.erase(iter);
+	return true;
+}
+
+bool RoutingTable::call_route_by_sp(const struct routing_table_route_entry_sp_info& route_entry_sp_info,
+									std::function<bool(struct routing_table_route_entry&)> callback) {
```

Comment:
Remove empty line before `callback` argument.

Reply:
It's to break an already long line (85)